src/core/Akka/Serialization/Serialization.cs

```diff
@@ -119,6 +119,26 @@ public Serialization(ExtendedActorSystem system)
             }
         }
 
+        private Serializer GetSerializerByName(string name)
+        {
+            if (name == null)
+                return null;
+
+            var serializersConfig = System.Settings.Config.GetConfig("akka.actor.serializers").AsEnumerable().ToList();
+            foreach (var kvp in serializersConfig)
+            {
+                if (kvp.Key.Equals(name))
+                {
+                    var serializerTypeName = kvp.Value.GetString();
+                    var serializerType = Type.GetType(serializerTypeName);
+
+                    var serializerId = SerializerIdentifierHelper.GetSerializerIdentifierFromConfig(serializerType, (ExtendedActorSystem)System);
+                    return GetSerializerById(serializerId);
```

Comment:
Because I want to leverage existing infrastructure as much as possible, without having to resort to creating a new map from name to id (which is possibly easier, but costs more memory, although small). I resolve the name to the serializer ID and use the existing map to return the instance based on that ID.

Im still not completely set on this implementation though. I might just go with a seperate dict that maps name -> id which is initialised in the constructor like the other mappings are.

Impact of this method should be small though, as its only called once per type, after which caching takes over.